rust/agents/relayer/src/msg/gelato_submitter.rs

```diff
@@ -0,0 +1,107 @@
+use std::sync::Arc;
+
+use abacus_base::CachingInterchainGasPaymaster;
+use abacus_base::{chains::GelatoConf, InboxContracts};
+use abacus_core::db::AbacusDB;
+use tokio::task::JoinHandle;
+
+use eyre::Result;
+use tokio::sync::mpsc;
+use tracing::{info_span, instrument::Instrumented, Instrument};
+
+use super::SubmitMessageArgs;
+
+// TODO(webbhorn): Metrics data.
+// TODO(webbhorn): Pull in ForwardRequestOp logic from prior branch.
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub(crate) struct GelatoSubmitter {
+    // Source of messages to submit.
+    rx: mpsc::Receiver<SubmitMessageArgs>,
+
+    // Interface to Inbox / InboxValidatorManager on the destination chain.
+    // Will be useful in retry logic to determine whether or not to re-submit
+    // forward request to Gelato, if e.g. we have confirmation via inbox syncer
+    // that the message has already been submitted by some other relayer.
+    inbox_contracts: InboxContracts,
+
+    // Contract tracking interchain gas payments for use when deciding whether
+    // sufficient funds have been provided for message forwarding.
+    interchain_gas_paymaster: Option<Arc<CachingInterchainGasPaymaster>>,
```

Comment:
what I said in `SerialSubmitter` also applies here

Reply:
Done.